src/Illuminate/View/Compilers/BladeCompiler.php

```diff
@@ -662,6 +662,17 @@ protected function compileElseif($expression)
     }
 
     /**
+     * Compile the has statements into valid PHP.
+     *
+     * @param  string  $expression
+     * @return string
+     */
+    protected function compileHas($expression)
+    {
+        return "<?php if(!empty(trim(\$__env->yieldContent{$expression}))): ?>";
```

Comment:
You may use $__env->hasSection to determine whether a section exists.

Actually, yieldContent does check the section is set